app/src/main/java/com/aptoide/uploader/apps/PackageManagerInstalledAppsProvider.java

```diff
@@ -40,6 +40,19 @@ public PackageManagerInstalledAppsProvider(PackageManager packageManager, Schedu
         .subscribeOn(scheduler);
   }
 
+  @Override public Single<InstalledApp> getInstalledApp(String packageName) throws PackageManager.NameNotFoundException {
+    return Single.just(packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA))
```

Comment:
What Single.just does is just creating the variable and then apply just in it, meaning that Single.just(x.tmp) is the same as String tmp = x.getTmp(); Single.just(tmp)
This means that the creation of tmp (in your case, packagemanager.getApplicationInfo) will happen before Rx is involved, which means that the exception will have to be handled before, that is why you had to add the throws ..
check it here: https://stackoverflow.com/questions/44992124/single-just-exception

If you use .fromCallable(()-> packagemanager...) you won't have to add the throws.
Sorry i should have probably told you earlier.